src/filesystem.c

```diff
@@ -241,53 +241,207 @@ rcutils_mkdir(const char * abs_path)
 size_t
 rcutils_calculate_directory_size(const char * directory_path, rcutils_allocator_t allocator)
```

Comment:
I understand that it is hard to make API changes across repositories, but I do think we should do that here.  In point of fact, this API has the same problems as the new `rcutils_calculate_directory_size_with_recursion` had, namely:

1.  You can't tell if the return value is 0 because the call failed, or because there were just no files in the directory.
1.  On 32-bit, you can only return a size up to 4GB, which is pretty small.

So I think we should change this signature to:

```
rcutils_ret_t
rcutils_calculate_directory_size(const char *directory_path, uint64_t *size, rcutils_allocator_t allocator)
```
And also update rosbag2 to use that signature as well.  It won't pass the PR job until we release both of them, but that's OK.  We'll run normal CI (that's why we have it), and then I can merge both and do a release of both at the same time.

Reply:
Okay. I will update rcutils_calculate_directory_size().  
And update rosbag2' codes since rcutils_calculate_directory_size() is changed.   

>   We'll run normal CI (that's why we have it)

I don't know this `normal CI`. What is the difference with the current system ？

Reply:
> I don't know this `normal CI`. What is the difference with the current system ？

Sorry, I should have been more clear.

We actually have two separate types of CI.  When you open a PR (and anytime you update it), the "Rpr" job runs on https://build.ros2.org.  You can see the output of that down below in the "Checks" section of this PR.  The PR job takes all of the current Rolling packages that this package depends on, installs them, checks this branch out, and then does a build-and-test.  This is great as it is automatic, and gives us fairly quick feedback on small issues like compiler warnings, linter warnings, unit tests, etc.  But it does have some problems:

1.  It won't test packages that depend on this package.  So in this case, the PR job succeeded, but if we merge this and then try to build rosbag2, it would fail because of the changed API.
1.  If there are API changes in packages that this package depends on, it will just fail.  There is no way to do a coordinated test across several repositories.
1.  It only tests Linux, not the other Tier-1 platforms.

For all of those reasons, we also have https://ci.ros2.org, where we can test all of the things above.

Therefore, for this PR, once we are happy with the code and it is approved, I can run a full test on ci.ros2.org to get cross-platform tests going, and with all packages that depend on this one.

Does that all make some sense?

Reply:
Thank you so much for your detailed explanation.  
I get it.  
I try to login https://ci.ros2.org and I think normal user have no permission to execute it.   
So you will help to execute it. Right ?

Reply:
> So you will help to execute it. Right ?

Yes, exactly.  Once I approve, I'll go ahead and run the job on https://ci.ros2.org.